client/test/error-telemetry.test.js

```diff
@@ -0,0 +1,48 @@
+jest.mock('fs');
+
+const assert         = require('assert');
+const fs             = require('fs');
+const ErrorTelemetry = require('../lib/error-telemetry');
+const mkdirp         = require('mkdirp');
+const logger = require('winston');
+
+describe('Error Telemetry Logging', () => {
+  beforeEach(() => {
+    /* Make sure memfs is flushed every time */
+    fs.volume.reset();
+  });
+
+  describe('setup() call', () => {
+    it('should not return a Promise', () => {
```

Comment:
in your case it's "should return a Promise"

but I am not sure that returning a Promise is a good idea in this case (in the current version of the code) because you are not waiting for other Promise, your code add listeners. If you simply return void, the listeners will still "live" and do the stuff you want.

In general, you "async" a method if you want to use some "await" calls inside its body.